src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -82,6 +154,43 @@ public void SingleProcess_TryWaitMultipleTimesBeforeCompleting()
             Assert.True(p.WaitForExit(0));
         }
 
+        [Fact]
+        public async Task SingleProcess_TryWaitAsyncMultipleTimesBeforeCompleting()
+        {
+            Process p = CreateProcessLong();
+            p.Start();
+
+            // Verify we can try to wait for the process to exit multiple times
+            using (var cts = new CancellationTokenSource(millisecondsDelay: 0))
+            {
+                var token = cts.Token;
+                for (int i = 0; i < 2; i++)
+                {
+                    OperationCanceledException ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => p.WaitForExitAsync(token));
+                    Assert.Equal(token, ex.CancellationToken);
+                    Assert.False(p.HasExited);
+                }
+            }
+
+            // Then wait until it exits and concurrently kill it.
+            // There's a race condition here, in that we really want to test
+            // killing it while we're waiting, but we could end up killing it
+            // before hand, in which case we're simply not testing exactly
+            // what we wanted to test, but everything should still work.
+            _ = Task.Delay(10).ContinueWith(_ => p.Kill());
+
+            using (var cts = new CancellationTokenSource(WaitInMS))
+            {
+                await p.WaitForExitAsync(cts.Token);
+                Assert.True(p.HasExited);
+            }
+            using (var cts = new CancellationTokenSource(millisecondsDelay: 0))
```

Comment:
As noted earlier, this is just synchronously creating an already canceled token.

Reply:
Fixed!